src/components/VoteArticle/VoteArticle.jsx

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from "react";
+import { updateArticleVotes } from "../../api";
+import { Button } from "react-bootstrap";
+import { Badge } from "react-bootstrap";
+
+const VoteArticle = ({ votes, article_id }) => {
+  const [presentVotes, setPresentVotes] = useState(votes);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setPresentVotes(votes);
+  }, [votes]);
+
+  const handleNewVote = () => {
+    updateArticleVotes(article_id, 1).catch(() => {
+      setPresentVotes((currentVotes) => currentVotes - 1);
+    });
+    setPresentVotes((currentVotes) => currentVotes + 1);
+  };
+
+  const handleRemoveVote = () => {
+    updateArticleVotes(article_id, 1).catch(() => {
+      setPresentVotes((currentVotes) => currentVotes + 1);
+    });
+    setPresentVotes((currentVotes) => currentVotes - 1);
+  };
+
+  if (isError) {
```

Comment:
As the error state is not set, the optimistic update resets the vote count, but the user isn’t informed about the need to retry.